charts/common-app/templates/_deployment.yaml

```diff
@@ -0,0 +1,278 @@
+                {{- if .Values.jmx.enabled }}
+                -Djava.rmi.server.hostname={{ .Values.jmx.host }}
+                -Dcom.sun.management.jmxremote.rmi.port={{ .Values.jmx.port }}
+                {{- end }}
+                -XX:MaxRAMPercentage={{ printf "%.1f" .Values.javaOptions.maxRAMPercentage }}
+                {{ .Values.javaOptions.others | join " " }}'
+        {{- if or (.Values.files) (and .Values.persistence.enabled .Values.statefulSet) (.Values.secretFilesRefs) }}
+          volumeMounts:
+            {{- range $key, $value := .Values.files }}
+            - name: config
+              mountPath: {{ printf "%s/%s" $value.mountPath $key | quote }}
+              subPath: {{ $key | quote }}
+            {{- end }}
+            {{- range .Values.secretFilesRefs }}
+            - name: {{ .volume }}
+              mountPath: {{ .mountPath }}
+              {{- if .readOnly }}
+              readOnly: true
+              {{- end }}
+              {{- if .subPath}}
+              subPath: {{.subPath }}
+              {{- end }}
+            {{- end }}
+            {{- if and .Values.persistence.enabled .Values.statefulSet }}
+            - name: datadir
+              mountPath: /tmp/kafka-streams
+            {{- end }}
+        {{- end }}
+        {{- if or (.Values.jmx.enabled) (.Values.ports) }}
+          ports:
+            {{- range .Values.ports }}
+            - containerPort: {{ .containerPort }}
+              name: {{ .name | quote }}
+              {{- if .protocol }}
+              protocol: {{ .protocol | quote }}
+              {{- end }}
+            {{- end }}
+            {{- if .Values.jmx.enabled }}
+            - containerPort: {{ .Values.jmx.port }}
+              name: jmx
+            {{- end }}
+        {{- end }}
+          {{- if .Values.livenessProbe }}
+          livenessProbe:
+          {{- .Values.livenessProbe | toYaml | nindent 12 }}
+          {{- end }}
+          {{- if .Values.readinessProbe }}
+          readinessProbe:
+          {{- .Values.readinessProbe | toYaml | nindent 12 }}
+          {{- end }}
+      {{- if .Values.prometheus.jmx.enabled }}
+        - name: prometheus-jmx-exporter
+          image: "{{ .Values.prometheus.jmx.image }}:{{ .Values.prometheus.jmx.imageTag }}"
+          imagePullPolicy: "{{ .Values.prometheus.jmx.imagePullPolicy }}"
+          args:
+            - {{ .Values.prometheus.jmx.port | quote }}
+            - /etc/jmx-{{ include "common-app.name" . }}/jmx-kafka-{{ include "common-app.name" . }}-prometheus.yml
+          env:
+            - name: JAVA_TOOL_OPTIONS
+              value: "-XX:MaxRAMPercentage=90.0"
+          ports:
+            - containerPort: {{ .Values.prometheus.jmx.port }}
+              name: prometheus
+          resources:
+{{ toYaml .Values.prometheus.jmx.resources | indent 12 }}
+          volumeMounts:
+            - name: jmx-config
+              mountPath: /etc/jmx-{{ include "common-app.name" . }}
```

Comment:
Mount path should also be static. Maybe `/etc/jmx-kafka-app`. Before it was `/etc/jmx-streams-app`